handout/forms/form-builder/form-builder_custom_validation.md

```diff
@@ -0,0 +1,102 @@
+# FormBuilder Custom Validation
+
+As useful as the built-in validators are, it is very useful to be able to include your own.
+Angular 2 allows you to do just that, with minimal effort.
+
+Let's assume we are using the same Login Form, but now we also want to test that our password has an exclamation mark somewhere in it.
+
+_app/login-form.component.ts_
+```ts
+function hasExclamationMark (input: FormControl) {
+  const hasExclamation = input.value.indexOf('!') >= 0;
+
+  return hasExclamation ? null : { needsExclamation: true };
+}
+
+
+// ...
+this.password = new FormControl('', [
+  Validators.required,
```

Comment:
May be useful to have a short section covering some of the out of the box validators that angular 2 provides. I'm not sure if these are documented, but they can be seen in the source code :) 

https://github.com/angular/angular/blob/master/modules/%40angular/forms/src/validators.ts

I guess pattern is the one we didn't really cover. But having them in a clear list is also nice. 

Also interesting to note that pattern kind of does the same thing as the `!` example